acceptance/tests/provider/package/yum.rb

```diff
@@ -0,0 +1,104 @@
+test_name "yum provider should install, uninstall, upgrade, downgrade, and use semantic versioning for ensuring desired version" do
+  confine :to, :platform => /el/
+  tag 'audit:high'
+
+  require 'puppet/acceptance/common_utils'
+  extend Puppet::Acceptance::PackageUtils
+  extend Puppet::Acceptance::ManifestUtils
+
+  package = 'openjpeg2.x86_64'
+
+  lower_package_version = ''
+  higher_package_version = ''
+  sorted_available_versions = []
+
+  agents.each do |agent|
+    yum_command = 'yum'
+
+    step "Setup: Check if tests can be run on this box" do
+      on(agent, "rpm -q #{package}", :acceptable_exit_codes => [1]) do |result|
+        skip_test "package #{package} already installed on this box" if result =~ /package #{package} is not installed/
+      end
+    end
+
+    step "Setup: Get two available package versions" do
+      on(agent, "yum list #{package} --showduplicates | sed -e '1,/Available Packages/ d' | awk '{print $2}'") do |result|
+        sorted_available_versions = result.stdout.split("\n").sort!.reverse
+        if sorted_available_versions.count >=2
+          higher_package_version = sorted_available_versions[0]
+          lower_package_version = sorted_available_versions[1]
+        end
```

Comment:
we are relying on ruby string sort here to decide which packages is higher. I think relying on this could cause failures in the future if new package versions appear.